uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/sickness-leave-request-instance_test.js

```diff
@@ -0,0 +1,119 @@
+define([
+  'mocks/helpers/helper',
+  'mocks/apis/leave-request-api-mock',
+  'mocks/apis/option-group-api-mock',
+  'leave-absences/shared/models/instances/sickness-leave-request-instance',
+], function (helper) {
+  'use strict';
+
+  describe('SicknessRequestInstance', function () {
+    var expectedError, instance, LeaveRequestAPI, $provide, promise, requestData, $rootScope;
+
+    beforeEach(module('leave-absences.models.instances', 'leave-absences.mocks', function (_$provide_) {
+      $provide = _$provide_;
+    }));
+
+    beforeEach(inject(function (_LeaveRequestAPIMock_, _OptionGroupAPIMock_) {
+      $provide.value('LeaveRequestAPI', _LeaveRequestAPIMock_);
+      $provide.value('OptionGroup', _OptionGroupAPIMock_);
+    }));
+
+    beforeEach(inject([
+       '$rootScope', 'LeaveRequestAPI', 'SicknessRequestInstance',
+      function (_$rootScope_, _LeaveRequestAPI_, _SicknessRequestInstance_) {
+        instance = _SicknessRequestInstance_.init({}, false);
+        $rootScope = _$rootScope_;
+        LeaveRequestAPI = _LeaveRequestAPI_;
+
+        spyOn(LeaveRequestAPI, 'create').and.callThrough();
+        spyOn(LeaveRequestAPI, 'update').and.callThrough();
+        spyOn(LeaveRequestAPI, 'isValid').and.callThrough();
+      }
+    ]));
+
+    describe('init', function () {
+      it('sickness request', function () {
+        expect(instance).toBeDefined();
+      });
+    });
+
+    describe('create()', function () {
+      beforeEach(function () {
+        requestData = helper.createRandomSicknessRequest();
+        instance = instance.init(requestData, false);
+        promise = instance.create();
+      });
+
+      afterEach(function () {
+        //to excute the promise force an digest
+        $rootScope.$apply();
+      });
+
+      it('calls equivalent API method', function () {
+        promise.then(function () {
+          expect(LeaveRequestAPI.create).toHaveBeenCalledWith(jasmine.any(Object), 'sick');
+        });
+      });
+
+      describe('id field', function() {
+        it('is not appended to instance before API returns data', function() {
+          expect(instance.id).not.toBeDefined();
+        });
+
+        it('is appended to instance before API returns data', function() {
+          promise.then(function () {
+            expect(instance.id).toBeDefined();
```

Comment:
` expect(instance.id).toBeDefined();` can be removed. Because `expect(instance.id).toEqual(jasmine.any(String));` ,means the previous is also true.